packages/resource-deployment/scripts/function-app-create.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+set -euo pipefail
+
+exitWithUsageInfo() {
+    echo "
+Usage: $0 -s <subscription name or id> -r <resource group> -l <location>
+"
+    exit 1
+}
+
+export subscription=""
+export resourceGroupName=""
+export location=""
+export functionAppName
+export appInsightsName
+export storageAccountName
+
+# Set default ARM Function App template files
+templateFilePath="../templates/function-app-template.json"
+
+# Read script arguments
+while getopts "s:r:l:"  option; do
+    case $option in
+		s) subscription=${OPTARG} ;;
+		r) resourceGroupName=${OPTARG} ;;
+		l) location=${OPTARG} ;;
+		*) exitWithUsageInfo ;;
+		esac
+done
+
+if [ -z "$subscription" ] || [ -z "$resourceGroupName" ] || [ -z "$location" ]; then
+	echo "Either one of subscription, resourceGroupName or location is empty"
+	exitWithUsageInfo
+fi
+
+# Login to Azure account if required
+if ! az account show 1>/dev/null; then
+    az login
+fi
+
+#set the default subscription id
+az account set --subscription $subscription
+
+#Check for existing RG
+if [ $(az group exists --name $resourceGroupName) = false ]; then
+	echo "Resource group with name" $resourceGroupName "could not be found. Creating new resource group.."
+	az group create --name $resourceGroupName --location $location 1> /dev/null
+	echo "Resource group has been successfully created"
+else
+	echo "Using existing resource group..."
+fi
+
+#Start deployment
+echo "Deploying Function App using ARM template"
+export resourceName
+resources=$(az group deployment create \
+    --resource-group "$resourceGroupName" \
+    --template-file "$templateFilePath" \
+    --query "properties.outputResources[].id" \
+    -o tsv)
+
+
+. "get-resource-name-from-resource-paths.sh" -p "Microsoft.Web/sites" -r "$resources"
```

Comment:
If I understand correctly we calling this three times right? Is there a way we can store the response and parse it thrice. (Yes I am aware its idempotent, but itd be great not to call it thrice). If we aren't calling this thrice ignore this completely.

Reply:
That'd would need a refactor for get-resource-name-from-resource-paths.sh, we can do it in a different PR

Reply:
Actually resolved since we don't need to create resources anymore, we're gonna use the previously created resources